lib/webpay_interswitch/form_builder.rb

```diff
@@ -77,5 +87,20 @@ def sanitize_options
         @submit_button_text = @html_options.delete(:submit_button_text) || 'Make Payment'
       end
 
+      def primary_parameters_valid?
+        @txn_ref.present? && Integer(@amount.to_s) > 0
+      end
+
+      def secondary_parameters_valid?
+        datetime_parameters_valid?
+      end
+
+      def datetime_parameters_valid?
+        if @optional_parameters[DATETIME_PARAM].present?
+          DateTime.parse(@optional_parameters[DATETIME_PARAM].to_s)
```

Comment:
This would generate the exception when incorrect date is passed.
Instead use:
```
DateTime.parse(@optional_parameters[DATETIME_PARAM].to_s) rescue false
```